src/overlays/actors/ovl_En_Clear_Tag/z_en_clear_tag.c

```diff
@@ -61,25 +76,1001 @@ static ColliderCylinderInit D_809D5C40 = {
+                                   this->targetDirection.x);
+                    zWorldRotationTarget = Math_SmoothStepToS(&thisx->world.rot.y, yWorldRotationTarget, rotationScale,
+                                                              this->targetDirection.y, 0);
+                    Math_ApproachF(&this->targetDirection.x, xRotationTarget, 1.0f, 256.0f);
+                    this->targetDirection.y = this->targetDirection.x;
+                    if (ABS(zWorldRotationTarget) < 4096) {
+                        Math_ApproachS(&thisx->world.rot.z, 0, 15, this->targetDirection.z);
+                        Math_ApproachF(&this->targetDirection.z, 1280.0f, 1.0f, 256.0f);
+
+                        // Check if the Arwing should fire it's laser.
+                        if (!(this->timer & 3) && (Rand_ZeroOne() < 0.75f) &&
+                            (this->state == CLEAR_TAG_STATE_TARGET_LOCKED)) {
+                            this->shouldShootLaser = 1;
+                        }
+                    } else {
+                        zWorldRotationTarget = zWorldRotationTarget > 0 ? -9472 : 9472;
+                        Math_ApproachS(&this->actor.world.rot.z, zWorldRotationTarget, rotationScale,
+                                       this->targetDirection.z);
+                        Math_ApproachF(&this->targetDirection.z, 4096.0f, 1.0f, 512.0f);
+                    }
+                    this->actor.shape.rot = this->actor.world.rot;
+                    this->actor.shape.rot.x = -this->actor.shape.rot.x;
+
+                    // Update the Arwing's velocity.
+                    func_8002D908(&this->actor);
+                    this->actor.velocity.x += this->acceleration.x;
+                    this->actor.velocity.y += this->acceleration.y;
+                    this->actor.velocity.z += this->acceleration.z;
+                    Math_ApproachZeroF(&this->acceleration.x, 1.0f, 1.0f);
+                    Math_ApproachZeroF(&this->acceleration.y, 1.0f, 1.0f);
+                    Math_ApproachZeroF(&this->acceleration.z, 1.0f, 1.0f);
+
+                    // Fire the Arwing laser.
+                    if (this->shouldShootLaser) {
+                        this->shouldShootLaser = 0;
+                        Actor_Spawn(&globalCtx2->actorCtx, globalCtx2, ACTOR_EN_CLEAR_TAG, this->actor.world.pos.x,
+                                    this->actor.world.pos.y, this->actor.world.pos.z, this->actor.world.rot.x,
+                                    this->actor.world.rot.y, this->actor.world.rot.z, CLEAR_TAG_STATE_LASER);
+                    }
+                }
+
+            case CLEAR_TAG_STATE_CRASHING:
+                // Decrement the crashing timer.
+                if (this->crashingTimer != 0) {
+                    this->crashingTimer--;
+                }
+
+                // Update Arwing collider to better match a ground collision.
+                func_8002D7EC(&this->actor);
```

Comment:
I dont think this comment is accurate to whats going on. what does it mean for it to better match the ground?
setting focus doesnt seem related to that, and setting AC and AT are not related to that either

Reply:
It is accurate in my opinion. It's updating the collider to update in a way that makes the ground collision look more correct.

You're right, though, that all of it is accomplishing that goal.